intezer_sdk/endpoint_analysis.py

```diff
@@ -1,23 +1,50 @@
+import glob
+import json
+import os
+import pathlib
+from concurrent.futures import ThreadPoolExecutor
 from typing import List
 
+from intezer_sdk import consts
+from intezer_sdk._endpoint_analysis_api import EndpointScanApi
+from intezer_sdk.analysis import logger
 from intezer_sdk.api import IntezerApi
 from intezer_sdk.api import get_global_api
 from intezer_sdk.base_analysis import Analysis
+from intezer_sdk.consts import EndpointAnalysisEndReason
 from intezer_sdk.sub_analysis import SubAnalysis
 
+SCAN_TYPE_OFFLINE_ENDPOINT_SCAN = 'offline_endpoint_scan'
 
 class EndpointAnalysis(Analysis):
     """
     EndpointAnalysis is a class for analyzing endpoints. It is a subclass of the Analysis class and requires an API connection to Intezer.
     """
-    def __init__(self, api: IntezerApi = None):
+    def __init__(self,
+                 api: IntezerApi = None,
+                 scan_api: EndpointScanApi = None,
+                 metadata_dir: str = None,
+                 files_dir: str = None,
+                 fileless_dir: str = None,
+                 memory_modules_dir: str = None):
         """
         Initializes an EndpointAnalysis object.
 
         :param api: The API connection to Intezer.
         """
         super().__init__(api)
+        self._scan_api = scan_api
+        if metadata_dir:
+            files_dir = files_dir or os.path.join(metadata_dir, '../files')
```

Comment:
does `../` works natively on windows?

Reply:
Better to use `os.path.join(metadata_dir, '..', 'files')`